pkg/containerimagebuilderpusher/kaniko.go

```diff
@@ -88,6 +89,49 @@ func (k *Kaniko) BuildAndPushContainerImage(buildOptions *BuildOptions, namespac
 	return errors.New("Kaniko job has timed out")
 }
 
+func (k *Kaniko) GetOnbuildStages(onbuildArtifacts []runtime.Artifact) ([]string, error) {
+	onbuildStages := make([]string, len(onbuildArtifacts))
+
+	for _, artifact := range onbuildArtifacts {
+		if artifact.ExternalImage {
+			continue
+		}
+
+		onbuildDockerfileContents := fmt.Sprintf(`FROM %s
+ARG NUCLIO_LABEL
+ARG NUCLIO_ARCH
+`, artifact.Image)
+
+		onbuildStages = append(onbuildStages, onbuildDockerfileContents)
+	}
+
+	return onbuildStages, nil
+}
+
+func (k *Kaniko) TransformOnbuildArtifactPaths(onbuildArtifacts []runtime.Artifact) (map[string]string, error) {
+	artifactIndex := 0
+
+	stagedArtifactPaths := make(map[string]string)
+	for _, artifact := range onbuildArtifacts {
```

Comment:
why not use `artifactIndex` as returned from range?

Reply:
Some of the artifacts may actually be external images, so a dedicated index counter for onbuild stages is needed